flocker/node/agents/functional/test_ebs.py

```diff
@@ -69,15 +62,16 @@ def test_foreign_volume(self):
         Test that ``list_volumes`` lists only those volumes
         belonging to the current Flocker cluster.
         """
-        ec2_client = ec2_client_from_environment()
+        cls, kwargs = get_blockdeviceapi_args(ProviderType.aws)
```

Comment:
If test code is going to be calling `get_blockdeviceapi_args` directly, then that function should probably be raising `SkipTest` directly.